### DISCUSSION TITLE
What does 'Mixed' mean?

### DISCUSSION THREAD

**Discussion by llk2yq** [typegoose/typegoose#693] [Q&A]
## Versions

- System:  windows 
- NodeJS: 16.x
- Typescript: 4.6.3

- Compiler / Transpiler: tsc / ts-node / ts-jest 
- Typegoose(NPM): 9.8.0 
- mongoose: 6.3.0 
- mongodb: 4.x
- nestjs: 8.4.4 
---

Here is my code:
```ts
import { getModelForClass, prop } from '@typegoose/typegoose';

interface ITest {}

export class Test {
    @prop()
    test?: ITest;
}

export const TestModel = getModelForClass(Test);
```
Then  I got this warning:
`Setting "Mixed" for property "Test.test"`.

What does it mean?  And how to remove this warning?

**Comment by llk2yq** [comment]
I found [this](https://github.com/typegoose/typegoose/issues/34#issuecomment-530911003)

**Comment by hasezoey** [comment]
a related error would be [E017](https://typegoose.github.io/typegoose/docs/guides/error-warning-details#setting-mixed-is-not-allowed-e017) (can be configured to either warn(default), error or be silent for these "problems")

---

the `Mixed` in this case refers to `mongoose.Types.Mixed`, which basically is the `any`(typescript) type in mongoose, this type allows anything set to it, and does not validation on the type, and mostly this only happens when something went wrong, so typegoose warn by default that a property is `Mixed` at runtime

if you really mean to use `Mixed`, then see #620 and [`modelOptions.options.allowMixed`](https://typegoose.github.io/typegoose/docs/api/decorators/model-options/#allowmixed)

[also here is the mongoose documentation for `Mixed`](https://mongoosejs.com/docs/schematypes.html#mixed)

PS: i will transfer this issue to a discussion Q&A
